packages/kit/src/runtime/server/index.js

```diff
@@ -105,7 +105,7 @@ export async function respond(incoming, options, state = {}) {
 					});
 				}
 
-				const decoded = decodeURI(request.url.pathname).replace(options.paths.base, '');
+				const decoded = decodeURI(request.url.pathname).replace(options.paths.base, '') || '/';
```

Comment:
Should the similar line in `prerender.js` get the same treatment?

Separately, should these be adjusted to only do the replacement at the start of the string? Is it possible that `options.paths.base` won't appear at the beginning of the path but does (by chance) appear later in the string?

If we know that `request.url.pathname` will _always_ begin with `options.paths.base`, can we just do `decoder(request.url.pathname).slice(options.paths.base.length) || '/'`? I'd expect this to be very slightly faster. And if it doesn't always begin with that, then I don't think this replacement is safe anyway.

Reply:
good points, will update. i don't think we can guarantee that `request.url.pathname.startsWith(options.paths.base)` — someone might be, i don't know, running several SvelteKit node apps from a single Express server

Reply:
updated